src/main/scala/is/hail/annotations/aggregators/RegionValueVirtualDSAggregator.scala

```diff
@@ -0,0 +1,52 @@
+package is.hail.annotations.aggregators
+import is.hail.annotations.RegionValueBuilder
+import is.hail.annotations._
+import is.hail.expr.types.{TArray, TFloat64, TTuple}
+import org.apache.spark.sql.Row
+
+object RegionValueVirtualDSAggregator {
+  val typ = TArray(TTuple(Array(TFloat64(), TFloat64())))
+}
+
+class RegionValueVirtualDSAggregator(xmin: Double, xmax: Double, ymin: Double, ymax: Double) extends RegionValueAggregator {
+  private var boxes = Set.empty[(Int, Int)]
+
+  def seqOp(r: Region, offset: Long, missing: Boolean) {
+    if (!missing) {
+      val row = UnsafeRow.read(TTuple(Array(TFloat64(), TFloat64())), r, offset).asInstanceOf[UnsafeRow]
+      val x = row.getDouble(0)
+      val y = row.getDouble(1)
+      val i = ((((x - xmin) / (xmax - xmin)) * 10).toInt, (((y - ymin) / (ymax - ymin)) * 10).toInt)
```

Comment:
I think this is good, but can be improved.  As Jackie says, 10 is way to coarse.  I was thinking ~500.  That gives 250K buckets, most of which will be absent.  Fine on a computer with gigabytes of memory.  You should put something like `val nDivisions = 500` in the RegionValueVirtualDSAggregator object and use it everywhere to make it easier to change.

Since toInt rounds down:

```
scala> (0.99).toInt
res0: Int = 0
```

The first division, after rescaling, represents coordinates [0, 1), the second [1, 2), ... and the last exactly N.  That isn't ideal, so I think we should have N-1 divisions and have the last division represent [N-1, N] (inclusive).

Now, if a coordinate (i, j) represents the boxes [i, i+1) by [j, j+1), the best representative of the box should be the center.  So I think when you convert back to Doubles, you should add 0.5.

Make sense?

Reply:
You might also consider adding a helper function on RegionValueVirtualDSAggregator to convert from two Doubles to the integer pair and calling it from both aggregators.